curvefs/src/client/client_operator.cpp

```diff
@@ -142,14 +142,14 @@ void RenameOperator::SetTxId(uint32_t partitionId, uint64_t txId) {
     metaClient_->SetTxId(partitionId, txId);
 }
 
-// TODO(Wine93): we should improve the check for whether a directory is empty
 CURVEFS_ERROR RenameOperator::CheckOverwrite() {
     if (dstDentry_.flag() & DentryFlag::TYPE_FILE_FLAG) {
         return CURVEFS_ERROR::OK;
     }
 
     std::list<Dentry> dentrys;
-    auto rc = dentryManager_->ListDentry(dstDentry_.inodeid(), &dentrys, 1);
+    auto rc = dentryManager_->CheckDirEmpty(dstDentry_.inodeid(),
```

Comment:
The purpose we add the `CheckDirEmpty` is to check wether the directory is empty, it no need to list entries, so the behavior maybe like:

```C++
bool yes;
auto rc = dentryManager_->CheckDirEmpty(ino, &yes);
if (rc != CURVEFS_ERROR::OK) {
    // handler error
} else if (yes) {
    // do something for empty directory
} else {
    // do something for non-empty directory
}
```

and maybe you should add a RPC to do it.

Reply:
OK, Got it.